packages/playground-examples/copy/ja/TypeScript/Type Primitives/Nullable Types.ts

```diff
@@ -0,0 +1,85 @@
+//// { order: 3, compiler: { strictNullChecks: false } }
+
+// JavaScriptでは、存在しない値を宣言する方法が2つあります。
+// TypeScriptでは、optionalやnullableな値を宣言する
+// 方法をさらにいくつか提供します。
+
+// 最初に、JavaScriptの基本型である
+// undefinedとnullの違いを見てみましょう。
+
+// Undefinedは値が見つからないときあるいは設定できない場合です。
+
+const emptyObj = {};
+const anUndefinedProperty: undefined = emptyObj["anything"];
+
+// Nullは値が意図的に欠如していることを
+// 意味します。
+
+const searchResults = {
+  video: { name: "LEGO Movie" },
+  text: null,
+  audio: { name: "LEGO Movie Soundtrack" },
+};
+
+// なぜundefinedを使わないのでしょう?
+// 主な理由としては、textが正しく含まれていることを確認できるからです。
+// もしtextがundefinedの場合、
+// 結果はtextが存在しないときと同じものになります。
+
+// これは、少し表面的に感じるかもしれません。
+// しかし、JSON文字列に変換したときに、
+// textがundefinedの場合、textは変換された文字列に含まれません。
+
+// Strict Null Types
+
+// TypeScript 2.0より前では、undefinedとnullは事実上、型システムから無視されていました。
+// これによって、TypeScriptのコーディング環境は
+// 型のないJavaScriptに近づいてしまっていました。
+
+// バージョン2.0にて、"strictNullChecks"というコンパイラフラグが追加されました。
+// このフラグをオンにすると、undefinedとnullが
+// コードフロー分析を通して対応すべき型として扱われるようになります。
+// (より詳細にはexample:code-flowを参照ください)
```

Comment:
```suggestion
// (より詳細には example:code-flow を参照ください)
```

リンク部分は前後にスペース入れないと認識されないのでは？